PlanningBiblio/PlanningBiblio
# MT29957: Password complexity

Asks the user to provide a robust enough password on password change.

The following rules are implemented:

 - the password must be at least 8 characters long.
 - the password must contain at least one digit.
 - the password must contain at least one capital letter.
 - the password must contain at least one lower-case letter.
 - the password must contain at least one special character, from the following
   list: !"#$%&'()*+,-./:;<=>?@[\]^_`{|}~
   ( comes from https://owasp.org/www-community/password-special-characters )

Also, the code has been refactored so:

 - the code checking the complexity is the same on client side and server side
   validation.
 - the UI for the password change is the same in "My Account" and Agent edition.

Test plan:

 - Log in with a user with the following rights : Gestion des agents.
 - Check that you can change the password of any agent, only if the complexity
   requirements are met.
 - Check that you cannot change the password without the "Gestion des agents"
   right.
 - Check that you can change your own password on the "My Account" page,
   regardless of your rights.
